File: python-samples/integer_string_conversion.py
from utils import ElapsedTimeMonitor

iterations = 20
batch_size = 10_000_000

global_time_monitor = ElapsedTimeMonitor()
iteration_duration_sum = 0

for n in range(0, iterations):
    iteration_time_monitor = ElapsedTimeMonitor()
    iteration_sum = 0

    for i in range(0, batch_size):
        s = str(i)
        iteration_sum += len(s)

    iteration_duration = iteration_time_monitor.get_elapsed_time_millis()
    iteration_duration_sum += iteration_duration

    print("iteration sum:", iteration_sum)
    print("iteration duration:", iteration_duration, "ms")

print("average iteration duration:", (iteration_duration_sum / float(iterations)), "ms")
print("global duration:", global_time_monitor.get_elapsed_time_millis(), "ms")

str_list = []

for i in range(0, batch_size):
    str_list.append(str(i))

global_time_monitor = ElapsedTimeMonitor()
iteration_duration_sum = 0

for n in range(0, iterations):
    iteration_time_monitor = ElapsedTimeMonitor()
    iteration_sum = 0

    for str_value in str_list:
        num = int(str_value)
        iteration_sum += num

    iteration_duration = iteration_time_monitor.get_elapsed_time_millis()
    iteration_duration_sum += iteration_duration

    print("iteration sum:", iteration_sum)
    print("iteration duration:", iteration_duration, "ms")

print("average iteration duration:", (iteration_duration_sum / float(iterations)), "ms")
print("global duration:", global_time_monitor.get_elapsed_time_millis(), "ms")

File: java-samples/src/main/java/posts/parthmistry/javasamples/IntegerStringConversion.java
package posts.parthmistry.javasamples;

import posts.parthmistry.javasamples.utils.ElapsedTimeMonitor;

import java.util.ArrayList;

public class IntegerStringConversion {

    public static void main(String[] args) {
        var iterations = 20;
        var batchSize = 10_000_000;

        var globalTimeMonitor = new ElapsedTimeMonitor();
        var iterationDurationSum = 0L;

        for (int n = 0; n < iterations; ++n) {
            var iterationTimeMonitor = new ElapsedTimeMonitor();
            long iterationSum = 0;

            for (int i = 0; i < batchSize; ++i) {
                var s = Integer.toString(i);
                iterationSum += s.length();
            }

            var iterationDuration = iterationTimeMonitor.getElapsedTimeMillis();
            iterationDurationSum += iterationDuration;

            System.out.println("iteration sum: " + iterationSum);
            System.out.println("iteration duration: " + iterationDuration + " ms");
        }

        System.out.println("average iteration duration: " + (iterationDurationSum / ((double) iterations)) + " ms");
        System.out.println("global duration: " + globalTimeMonitor.getElapsedTimeMillis() + " ms");
        System.out.println();

        var strList = new ArrayList<String>();

        for (int i = 0; i < batchSize; i++) {
            strList.add(Integer.toString(i));
        }

        globalTimeMonitor = new ElapsedTimeMonitor();
        iterationDurationSum = 0L;

        for (int n = 0; n < iterations; ++n) {
            var iterationTimeMonitor = new ElapsedTimeMonitor();
            long iterationSum = 0;

            for (var s : strList) {
                var num = Integer.parseInt(s);
                iterationSum += num;
            }

            var iterationDuration = iterationTimeMonitor.getElapsedTimeMillis();
            iterationDurationSum += iterationDuration;

            System.out.println("iteration sum: " + iterationSum);
            System.out.println("iteration duration: " + iterationDuration + " ms");
        }

        System.out.println("average iteration duration: " + (iterationDurationSum / ((double) iterations)) + " ms");
        System.out.println("global duration: " + globalTimeMonitor.getElapsedTimeMillis() + " ms");
    }

}
